File: 005/Python/problem005.py
import functools

def smallest_multiple(n):
    return functools.reduce(multiplier, range(1, n+1))

def multiplier(acc, i):
    if acc % i == 0:
        return acc
    else:
        return acc * i // gcd(acc, i)

def gcd(a, b):
    if b == 0:
       return a
    else:
       return gcd(b, a % b)

File: 005/Erlang/problem005.erl
-module(problem005). 
-export([smallest_multiple/1]).
-use(lists).

smallest_multiple(N) -> lists:foldr(fun (A, AccIn) -> multiplier(A, AccIn) end, 1, lists:seq(1, N)).

multiplier(I, Acc) when Acc rem I == 0 -> Acc;
multiplier(I, Acc) -> Acc * I div gcd(Acc, I).

gcd(A, B) when B == 0 -> A;
gcd(A, B) -> gcd(B, A rem B).
